Implement Contribute Page

harsh3dev opened the pull request:
<!-- Thanks for contributing to OWASP Nest!-->

<!-- Don't forget to link your PR to an existing issue if any.-->
Resolves #88 

<!-- Describe the big picture of your changes.-->

Uploading Recording 2024-12-22 184922.mp4…

- [x] Add Contribute page
- [x] Add tests for contribute page
- [x] Add Search and Pagination
- [x] Consistent with new code of displaying pages

<!-- Thanks again for your contribution!-->

kasya commented on frontend/src/components/Card.tsx:
```diff
@@ -99,7 +99,7 @@ const Card = ({
           </div>
           {projectName && (
-            <a href={projectLink} target="_blank" rel="noopener noreferrer">
+            <a href={projectLink} target="_blank" rel="noopener noreferrer" className="font-bold">
```
@harsh3dev Longer titles don't work with this layout right now: 
![Screenshot 2024-12-22 at 9 30 56 AM](https://github.com/user-attachments/assets/335b40a6-69f9-4b4a-9b84-1f1aea4c674b)
Could you update it and make sure that other pages still look good?

harsh3dev replied:
I will update this after updating the existing issues.
Thanks for reviewing.

kasya commented on frontend/src/components/Modal/Modal.tsx:
This new modal doesn't look like the one we had before. Can we make it as close as possible to what we had before for both dark and light theme?
Current modal view in dark theme:

![Screenshot 2024-12-22 at 9 33 22 AM](https://github.com/user-attachments/assets/186f51f5-5970-4b05-b816-c5651e462788)

Prev. modal view. You can reference it on our old backend port 8000:
![Screenshot 2024-12-22 at 9 33 53 AM](https://github.com/user-attachments/assets/23aabda1-01a7-4a22-8a39-0bdb617f2d2b)

harsh3dev replied:
Sure Ill update the modal

kasya commented on frontend/src/components/Card.tsx:
Also I noticed that the title on the card is now underneath the summary text. Could you update it to look like it was before - to have the project name underneath the title for the card?
![Screenshot 2024-12-22 at 9 38 03 AM](https://github.com/user-attachments/assets/d34985a3-b28d-464c-a5ff-83ba2a91ef55)

harsh3dev replied:
Okay, that'll be updated.

kasya commented on frontend/src/components/Modal/Modal.tsx:
The modal should be closed on clicking out of it. Let's update that too!
Also, I'm not sure I like the scrollable view in the modal. Can we just make it fit the full size of the available/needed screen if there are a lot of text. But if it doesn't fit the screen size - then we'll scroll?

harsh3dev replied:
Sure I'll implement that.

harsh3dev commented:
@kasya @arkid15r 
I have implemented the suggested changes and fixed the tests that were failing.

arkid15r commented:
> @kasya @arkid15r I have implemented the suggested changes and fixed the tests that were failing.

It seems the Markdown part is still missing. Let's work towards merging this one and improve it gradually.

kasya commented on frontend/src/components/Card.tsx:
```diff
@@ -98,11 +103,7 @@ const Card = ({
-            </a>
-          )}
+
           {(languages || (topics && topics.length > 0) || (social && social.length > 0)) && (
```
For issues card we also want to make sure we show topics. I believe they are already added here in the card itself, but I don't see any right now. Could you please check why is that? The topics should go under the languages labels. Screenshot of the same card at port 8000 and this PR:
![Screenshot 2024-12-23 at 8 27 56 AM](https://github.com/user-attachments/assets/3cab7d40-ed73-4691-92b8-17993c0d97eb)
![Screenshot 2024-12-23 at 8 28 00 AM](https://github.com/user-attachments/assets/3fa74070-395f-40f1-b594-62048357242a)

kasya commented on frontend/src/components/Modal/Modal.tsx:
This is better! I can see the markdown is still not working, but we can address that in a separate task! 👍🏼 

Another I thing I noticed with this modal is that the title and the text inside it are not exactly as they were before.
Can we make the title of the modal bolder, while reducing font-weight on the items inside the modal? Right now the `How to tackle it` thing has more weight then the title, which is not good. 
Also, can we make subtitle text muted?

Also some screenshots of current PR and modal at port 8000:
![Screenshot 2024-12-23 at 8 34 33 AM](https://github.com/user-attachments/assets/0f0ceb17-a420-4936-aae2-bafcbb2f3461)
![Screenshot 2024-12-23 at 8 34 39 AM](https://github.com/user-attachments/assets/8056c714-7194-4467-a81c-5058cd571d9e)

kasya commented on frontend/src/components/Modal/Modal.tsx:
Also when I'm clicking out of the Modal it does close, but also the webpage is kinda of jumping a little bit. Can we fix this?

https://github.com/user-attachments/assets/ce6c788d-b9c9-4780-baad-3cf0278b0224

kasya commented on frontend/src/components/Modal/Modal.tsx:
Also, can we increase the space between the title and the close button in the modal header?
![Screenshot 2024-12-23 at 8 47 39 AM](https://github.com/user-attachments/assets/ec063362-9ddb-4794-ada5-380856b7c9d8)

kasya reviewed (changes requested):
Nice updates! 👍🏼  A few comments:

kasya reviewed (changes requested):
Not related to this task probably, but I just noticed that the card doesn't have a white background anymore 😮 Can we update this? 
This is how they used to look like:
![Screenshot 2024-12-23 at 8 58 27 AM](https://github.com/user-attachments/assets/92288de2-ef03-4824-9e92-a6dcc98414e3)

harsh3dev commented:
@kasya updated the components based on the suggestions.
- show topics in issues 
- make modal title bold 
- make h2 inside modal normal font-weight 
- make subtitle text muted 
- increase the space between title and close button on the top
- make the card background white
- fix the stuttering of open/close modal
Also, I am planning to implement the Markdown in the following PR if that's okay. As I am still figuring out some errors on that.

arkid15r reviewed (changes requested):
@harsh3dev it's a bunch of great improvements, thank you!
Today we've merged https://github.com/OWASP/Nest/pull/247 please adjust your code for the new approach we use for Algolia requests.

Thank you!

kasya commented on frontend/src/index.css:
```diff
@@ -65,6 +65,52 @@ a {
 
-.navlink {
-  color: inherit;
+
```
Why have you removed the `navlink` class from there? 🤔 
It was responsible for links color in the Header component. They are not blue and not the dark grey color.
![Screenshot 2024-12-23 at 3 01 05 PM](https://github.com/user-attachments/assets/a7d80906-dd72-4ad3-89b6-30ae7c1deeac)

kasya commented on frontend/src/components/Card.tsx:
```diff
@@ -73,6 +73,11 @@ const Card = ({
         </div>
       </div>
+      {projectName && (
+        <a href={projectLink} target="_blank" rel="noopener noreferrer" className="font-bold">
```
With longer title I noticed that the meta data block gets separated from the edge of the card. If you look closely on this card the left border for the number of comments and date updated block is not touching the top of the card. It should not depend on the size of the title and always "connect" with the card.
Let me know if you need help with updating that too!
![Screenshot 2024-12-23 at 3 03 18 PM](https://github.com/user-attachments/assets/9391c329-172d-45d5-a99b-b88d74b45a78)

kasya reviewed (changes requested):
Thanks for addressing all those comments! Noticed a couple more things after these recent updates:

harsh3dev commented:
@arkid15r I have implemented all the requested changes, I have also added the new way of fetching data.
Although the tests after the changes were a bit of an obstacle but nevertheless I fixed them too.

https://github.com/user-attachments/assets/be759b45-70a9-498b-a000-a82c6a05d609

@kasya I also implemented the UI changes.
Please merge this one if it feels complete, I will handle the markdown and the minor UI changes if any in the upcoming PRs.

arkid15r commented on frontend/package.json:
Could you double check the added dependencies -- do we really need them?

arkid15r commented on frontend/src/pages/Contribute.tsx:
```diff
@@ -0,0 +1,79 @@
+      searchQuery={searchQuery}
+      onSearch={handleSearch}
+      onPageChange={handlePageChange}
+      searchPlaceholder="Search for OWASP Issues..."
```
```suggestion
      searchPlaceholder="Search for OWASP issues..."
```

kasya reviewed (approved):
@harsh3dev I pushed a couple of changes to address card styling issues that I noticed!

Nice work 👏🏼 Approved!

arkid15r reviewed (approved):
LGTM 👍
